lib/internal/console/constructor.js

```diff
@@ -518,14 +518,35 @@ function timeLogImpl(self, name, label, data) {
   }
   const duration = process.hrtime(time);
   const ms = duration[0] * 1000 + duration[1] / 1e6;
+
+  const formatted = formatTime(ms);
+
   if (data === undefined) {
-    self.log('%s: %sms', label, ms.toFixed(3));
+    self.log('%s: %s', label, formatted);
   } else {
-    self.log('%s: %sms', label, ms.toFixed(3), ...data);
+    self.log('%s: %s', label, formatted, ...data);
   }
   return true;
 }
 
+function formatTime(ms) {
+  let value = ms;
+  let unit = 'ms';
+
+  if (ms >= 3600000) {
+    value = ms / 3600000;
+    unit = 'h';
+  } else if (ms >= 60000) {
+    value = ms / 60000;
+    unit = 'min';
+  } else if (ms >= 1000) {
+    value = ms / 1000;
+    unit = 's';
+  }
+
+  return value.toFixed(3) + unit;
```

Comment:
i think we should always show the milliseconds

Reply:
I'm not convinced. even in case it's a few seconds, I don't think we need more than millisecond precision.

Reply:
if we don't include milliseconds can we remove the toFixed?

Reply:
`time` and `timeEnd` couple seems to be here to allow developers to get an idea of which time scale does a part of their code take to run. I think it's usefull to know if the code takes 500ms, 1, 1min or 1h to run but I don't see any case where the difference between 1h and 1.0000000000000001h is important. People that need high precision can still use `process.hrtime`.

Reply:
I am in favor of changing the representation in general but I think it would have been good to keep the milliseconds. I think it would have been ideal to visualize it as: `n1h, n2m, n3s and n4ms` (n1 - n4 as value) while each leading part would be optional as long as all former ones are zero. Right now you have to calculate in a fraction of an hour or minutes and that is not very intuitive.